services/openshiftbuilddeploy/src/index.js

```diff
@@ -388,7 +389,7 @@ const messageConsumer = async msg => {
       if (err.code == 404) {
         logger.info(`${openshiftProject}: RoleBinding ${openshiftProject}-lagoon-deployer-image-puller in ${openshiftPromoteSourceProject} does not exists, creating`)
         const promotionSourceRolebindingsPost = Promise.promisify(openshift.ns(openshiftPromoteSourceProject).rolebindings.post, { context: openshift.ns(openshiftPromoteSourceProject).rolebindings })
-        await promotionSourceRolebindingsPost({ body: {"kind":"RoleBinding","apiVersion":"v1","metadata":{"name":`${openshiftProject}-lagoon-deployer-image-puller`,"namespace":openshiftPromoteSourceProject},"roleRef":{"name":"system:image-puller"},"subjects":[{"name":"lagoon-deployer","kind":"ServiceAccount","namespace":openshiftProject}]}})
+        await promotionSourceRolebindingsPost({ body: {"kind":"RoleBinding","apiVersion":"v1","metadata":{"name":`${openshiftProject}-lagoon-deployer-image-puller`,"namespace":openshiftPromoteSourceProject},"roleRef":{"name":"system:image-puller"},"subjects":[{"name":"lagoon-deployer","kind":"ServiceAccount","namespace":openshiftProject},{"name":"default","kind":"ServiceAccount","namespace":openshiftProject}]}})
```

Comment:
good start! But I think we need to adapt this a bit. I see two problems:
1. you are using an existing rolebinding that is called `${openshiftProject}-lagoon-deployer-image-puller` but it includes more than what the name suggests (it includes also a role binding for the default serviceaccount)
2. this code is only executed if `${openshiftProject}-lagoon-deployer-image-puller` does not exist yet and therefore it will only be fixed for new promotion projects, even though we currently have promotion projects out there already and we want to fix them as well.

Because of this I suggest to create a new rolebinding to solve the two problems in one.

Reply:
I've created a new RoleBinding: `${openshiftProject}-lagoon-default-image-puller`
See latest commit

Reply:
746ddd5a78c1aae27f76f22d5dcab5dd2beea6be